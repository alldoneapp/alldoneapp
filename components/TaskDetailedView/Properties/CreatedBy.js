import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

import Creator from './Creator'
import Icon from '../../Icon'
import styles, { colors } from '../../styles/global'
import { translate } from '../../../i18n/TranslationService'

export default function CreatedBy({ createdDate, creator }) {
    return (
        <View style={localStyles.container}>
            <View style={{ marginRight: 8 }}>
                <Icon name="user" size={24} color={colors.Text03} />
            </View>
            <Text style={[styles.subtitle2, { color: colors.Text03 }]}>{translate('Owner')}</Text>
            <View style={{ marginLeft: 'auto' }}>
                <Creator createdDate={createdDate} creator={creator} />
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        maxHeight: 56,
        minHeight: 56,
        height: 56,
        paddingLeft: 8,
        paddingVertical: 8,
        alignItems: 'center',
    },
})
