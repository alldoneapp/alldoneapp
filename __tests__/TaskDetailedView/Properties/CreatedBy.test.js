import React from 'react'
import CreatedBy from '../../../components/TaskDetailedView/Properties/CreatedBy'

import renderer from 'react-test-renderer'

describe('CreatedBy component', () => {
    describe('CreatedBy snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer
                .create(
                    <CreatedBy createdDate={541235648} creator={{ photoURL: 'https://', displayName: 'Master Yoda' }} />
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
