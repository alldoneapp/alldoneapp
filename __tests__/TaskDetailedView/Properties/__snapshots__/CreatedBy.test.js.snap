// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatedBy component CreatedBy snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "maxHeight": 56,
      "paddingLeft": 11,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 11,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 20,
            "opacity": 1,
          }
        }
      >
        î¨¥
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#04142F",
          "fontFamily": "Roboto-Medium",
          "fontSize": 14,
          "letterSpacing": 0.16,
          "lineHeight": 22,
        },
        Object {
          "color": "#8A94A6",
        },
      ]
    }
  >
    Created by
  </Text>
  <View
    style={
      Object {
        "marginLeft": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingLeft": 11,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "marginRight": 12,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://",
            }
          }
          style={
            Object {
              "borderRadius": 100,
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Regular",
                "fontSize": 16,
                "letterSpacing": 0.32,
                "lineHeight": 24,
              },
              Object {
                "color": "#04142F",
              },
            ]
          }
        >
          Master Yoda
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#04142F",
                "fontFamily": "Roboto-Regular",
                "fontSize": 12,
                "letterSpacing": 0.48,
                "lineHeight": 20,
              },
              Object {
                "color": "#8A94A6",
              },
            ]
          }
        >
          on the 01/07/1970
        </Text>
      </View>
    </View>
  </View>
</View>
`;
